src/e2e-test/java/io/cdap/cdap/ui/stepsdesign/PipelineTriggers.java

```diff
@@ -0,0 +1,146 @@
+  public void openInboundTriggerAndSetAComplexTrigger(String sourcePipeline) {
+    ElementHelper.clickOnElement(Helper.locateElementByTestId("inbound-triggers-toggle"));
+    WebElement triggerNameInputField = Helper.locateElementByCssSelector(
+      "div[data-testid='trigger-name-text-field'] input"
+    );
+    ElementHelper.clearElementValue(triggerNameInputField);
+    ElementHelper.sendKeys(triggerNameInputField, complexTriggerName);
+
+    ElementHelper.clickOnElement(Helper.locateElementByTestId(sourcePipeline + "-collapsed"));
+    ElementHelper.clickOnElement(Helper.locateElementByTestId(sourcePipeline + "-trigger-config-btn"));
+
+    selectRuntimeArg(getSourceRuntimeArgElement(0), "source_path");
+    selectRuntimeArg(getTargetRuntimeArgElement(0), "source_path");
+    selectRuntimeArg(getSourceRuntimeArgElement(1), "sink_path");
+    selectRuntimeArg(getTargetRuntimeArgElement(1), "sink_path");
+
+    ElementHelper.clickOnElement(Helper.locateElementByTestId("configure-and-enable-trigger-btn"));
+    ElementHelper.clickOnElement(Helper.locateElementByTestId("enable-group-trigger-btn"));
+
+    ElementHelper.clickOnElement(Helper.locateElementByTestId(complexTriggerName + "-collapsed"));
+    ElementHelper.clickOnElement(Helper.locateElementByTestId(sourcePipeline + "-view-payload-btn"));
+
+    WebElement sourceArg0 = getSourceRuntimeArgElement(0);
+    Assert.assertEquals("source_path", sourceArg0.getAttribute("value"));
+    WebElement sourceArg1 = getSourceRuntimeArgElement(1);
+    Assert.assertEquals("sink_path", sourceArg1.getAttribute("value"));
+
+    WebElement targetArg0 = getTargetRuntimeArgElement(0);
+    Assert.assertEquals("source_path", targetArg0.getAttribute("value"));
+    WebElement targetArg1 = getTargetRuntimeArgElement(1);
+    Assert.assertEquals("sink_path", targetArg1.getAttribute("value"));
+
+    ElementHelper.clickOnElement(Helper.locateElementByTestId("close-payload-config-modal"));
+  }
+
+  @Then("Cleanup pipeline {string}")
+  public void cleanupPipeline(String pipelineName) {
+    Helper.cleanupPipelines(pipelineName);
+  }
+
+  @Then("Visit pipeline {string}")
+  public void visitPipeline(String pipelineName) {
+    Helper.gotoDeployedPipeline(pipelineName);
+  }
+
+  @Then("Ensure outbound trigger to {string} exists")
+  public void ensureOutboundTriggerExists(String triggeredPipeline) {
+    ElementHelper.clickOnElement(Helper.locateElementByTestId("outbound-triggers-toggle"));
+    ElementHelper.clickOnElement(Helper.locateElementByTestId(triggeredPipeline + "-triggered-collapsed"));
+    // The animation triggered on the line above takes some time to complete,
+    // so wait for text at the bottom to be visible
+    WebElement outboundTrigger = Helper.locateElementByTestId(triggeredPipeline + "-triggered-expanded");
+    SeleniumDriver.getWaitDriver(3).until(ExpectedConditions.textToBePresentInElement(outboundTrigger, "Succeeds"));
```

Comment:
Have you tried `WaitHelper.waitForElementToBeDisplayed` or `WaitHelper.waitForElementToBePresent`?

Reply:
Yes, those were not sufficient.